crates/ide/src/hover.rs

```diff
@@ -4446,4 +4515,71 @@ fn foo() -> Option<()> {
                 ```"#]],
         );
     }
+
+    #[test]
+    fn hover_deref_expr() {
+        check_hover_range(
+            r#"
+//- minicore: deref
+use core::ops::Deref;
+
+struct DerefExample<T> {
+    value: T
+}
+
+impl<T> Deref for DerefExample<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.value
+    }
+}
+
+fn foo() {
+    let x = DerefExample { value: 0 };
+    let y: i32 = $0*x$0;
+}
+"#,
+            expect![[r#"
+                ```text
+                Type:                            i32
+                Dereferenced from: DerefExample<i32>
```

Comment:
Same as in the other review here.
```suggestion
                Dereferenced from: DerefExample<i32>
                To Type:                         i32
```

Looks good to me otherwise 👍
bors d+